api/v1beta1/cluster_types.go

```diff
@@ -440,11 +440,17 @@ type ClusterStatus struct {
 	// FailureReason indicates that there is a fatal problem reconciling the
 	// state, and will be set to a token value suitable for
 	// programmatic interpretation.
+	//
+	// Deprecated: This field is deprecated and is going to be removed in the next apiVersion. Please see docs/proposals/20240916-improve-status-in-CAPI-resources.md for more details.
```

Comment:
Let's replace the relative paths with this `https://github.com/kubernetes-sigs/cluster-api/blob/main/docs/proposals/20240916-improve-status-in-CAPI-resources.md`

Relative links don't work well lin CRD YAML